Created  Header Section & adding slider and also Created Card-section

akshayyelvankar opened the pull request:
today working on Navbar,Header Section and Card Section.

akshayyelvankar committed:
Created  Header Section & adding slider and also Created Card-section

ankushbkadam commented on src/App.css:
```diff
@@ -32,7 +32,22 @@
+
+/* Header-slider */
+
+.carousel-item img {
```
never modify carousel, make a new class with these styles and use that class

akshayyelvankar replied:
changes done i add new class is carousel-contain

ankushbkadam replied:
do we really need to add manual css? can we avoid adding css and use bootstrap instead?

akshayyelvankar replied:
manual css is needed because in Converter component background effects are come from this css file.

ankushbkadam replied:
lets see if we really need it, lets just focus on home page

akshayyelvankar replied:
ok

ankushbkadam commented on src/App.css:
```diff
@@ -32,7 +32,22 @@
+}
+
+/* Card-Section */
+.card {
```
here as well, never modify card, make a class with these styles and use that class

akshayyelvankar replied:
changes done added new class card-bg

ankushbkadam replied:
do we really need? can you check if any alternative present in bootstrap

akshayyelvankar replied:
removing manual class card-bg and added bootstarp class bg-info.

ankushbkadam commented on src/Components/Home/Home-page.tsx:
```diff
@@ -1,8 +1,47 @@
-    )
+const Home = () => {
+   return (
+      <div className="">
```
why empty className? remove if not needed

ankushbkadam replied:
expecting reply to each and every comment

akshayyelvankar replied:
changes done.

ankushbkadam commented on src/Components/Home/Home-page.tsx:
```diff
@@ -1,8 +1,47 @@
+               </div>
+               <div className="carousel-inner">
+                  <div className="carousel-item active">
+                     <img src="./Images/p1.jpg" className="d-block w-100   " alt="pic-1" img-fluid />
```
remove heavy and full hd images, observe app is getting slow due to images!
If we want to use images convert them to lower resolution

akshayyelvankar replied:
changes done images files size are compress.

ankushbkadam replied:
can we use remote images

ankushbkadam commented on src/Components/Home/Home-page.tsx:
```diff
@@ -1,8 +1,47 @@
+   return (
+      <div className="">
+
+         <header className="container-fluid p-0">
```
this is not header, move this header section to another file and rename as carousel component

akshayyelvankar replied:
changes done.

ankushbkadam replied:
change filename to index.tsx

akshayyelvankar replied:
changes done renaming file home.tsx to index.tsx

ankushbkadam commented on src/Components/Home/Home-page.tsx:
```diff
@@ -1,8 +1,47 @@
+       
+         <section className=" container card-section mt-5">
+            <h2>Card Menu</h2>
+            <div className="card w-100 ">
```
check if we really need w-100? remove if not needed

akshayyelvankar replied:
changes done.

ankushbkadam commented on src/Components/Home/Home-page.tsx:
```diff
@@ -1,8 +1,47 @@
+         </header>
+       
+         <section className=" container card-section mt-5">
+            <h2>Card Menu</h2>
```
'Card Menu' ? 
give some meaningful title, may be 'Our Apps' ? or any which matches our requirement

akshayyelvankar replied:
changes done

ankushbkadam commented on src/Components/Navbar/Navbar.tsx:
```diff
@@ -5,7 +5,7 @@ import { ROUTES } from "../../Constants/routeConstants";
 const Navbar = () => {
     return (
         <div>
```
remove unnecessary div here

akshayyelvankar replied:
changes done

ankushbkadam commented on src/Components/Navbar/Navbar.tsx:
```diff
@@ -17,7 +17,7 @@ const Navbar = () => {
                             </li>
                             <li className="nav-item">
-                                <a className="nav-link" href={ROUTES.ABOUT}>About</a>
+                                <a className="nav-link active" href={ROUTES.ABOUT}>About</a>
```
both Home and About are active at same time? will it make any sense?
only active menu should have active right?

akshayyelvankar replied:
changes done.

ankushbkadam replied:
Home seems always active, do we need like that?

akshayyelvankar replied:
removed active class.

ankushbkadam commented on src/Components/Home/Home-page.tsx:
```diff
@@ -1,8 +1,47 @@
+               <div className="card-body p-5">
+                  <h5 className="card-title">Card title</h5>
+                  <p className="card-text">With supporting text below as a natural lead-in to additional content.</p>
+                  <a href="#" className="btn btn-primary">Go to Converter</a>
```
remove #, use constant for routes here

akshayyelvankar replied:
changes done.

ankushbkadam replied:
can we rename converter to a meaningful name? like which converter?

akshayyelvankar replied:
changes done renaming converter component to Rgb-to-Hex component.

akshayyelvankar committed:
modified

akshayyelvankar committed:
changes modified

akshayyelvankar committed:
title updated

akshayyelvankar committed:
changes done

akshayyelvankar committed:
navbar modified

akshayyelvankar committed:
added new Carsoule component

akshayyelvankar committed:
changes modified

akshayyelvankar committed:
images size are converted to low resolution and also add new class carousel-contain

akshayyelvankar committed:
changes done added new class card-bg

akshayyelvankar committed:
Merge branch 'development' of https://github.com/ankushbkadam/reactjs-tools into epic/home-page

akshayyelvankar committed:
renaming home to index.tsx removing some manual css clases

akshayyelvankar committed:
resloving some change like removing active class etc

akshayyelvankar committed:
changes done

akshayyelvankar committed:
changes done

akshayyelvankar committed:
done

akshayyelvankar committed:
done

akshayyelvankar committed:
flie name to index.tsx,header tag are removed

akshayyelvankar committed:
added Constant for Image in  Carousel component,issue reslove in active navbar menu,renaming all files to index.tsx

akshayyelvankar committed:
logo added in Navbar

akshayyelvankar committed:
removing unwanted img file

akshayyelvankar commented:
added Constant for Image in  Carousel component , issue resloved in active navbar menu , renaming all files to index.tsx and card section are well set in center and added logo in Navbar.

ankushbkadam commented on src/App.tsx:
```diff
@@ -1,24 +1,25 @@
 function App() {
   return (
-    <div className="App">
+    <div className="App d-flex flex-column min-vh-100">
```
why global flex? Remove global styles from here and do them in required components
This will get applied to each and every components in this case!

ankushbkadam commented on src/App.tsx:
```diff
@@ -1,24 +1,25 @@
         </Routes>
       </div>
-      <Footer />
+       <Footer /> 
```
why empty spaces added here?

akshayyelvankar replied:
Unwanted Spaces are Removed.

ankushbkadam commented on src/Components/Carousel/index.tsx:
```diff
@@ -0,0 +1,35 @@
+                    <button type="button" data-bs-target="#carouselExampleIndicators" data-bs-slide-to="2" aria-label="Slide 3"></button>
+                </div>
+                <div className="carousel-inner">
+                    <div className="carousel-item carousel-contain active">
```
why reparative code? can we reuse code?

ankushbkadam commented on src/Components/Home/index.tsx:
```diff
@@ -0,0 +1,20 @@
+import { ROUTES } from "../../Constants/routeConstants";
+import Carousel from "../Carousel";
+const Home = () => {
```
we should give one line space after all imports and component

akshayyelvankar replied:
I added one line of space after all imports and component.

ankushbkadam commented on src/Components/Home/index.tsx:
```diff
@@ -0,0 +1,20 @@
+import Carousel from "../Carousel";
+const Home = () => {
+   return (
+      <div>
```
make use of fragments instead div's

akshayyelvankar replied:
I Added empty fragments.

ankushbkadam commented on src/Components/Home/index.tsx:
```diff
@@ -0,0 +1,20 @@
+const Home = () => {
+   return (
+      <div>
+          <Carousel/>
```
indent

ankushbkadam commented on src/Components/Home/index.tsx:
```diff
@@ -0,0 +1,20 @@
+         <section className="container card-section pt-5 mb-4">
+            <h2>Our App Menu</h2>
+            <div className="card w-50 bg-info rounded mx-auto">
+               <div className="card-body" >
```
why empty space before closing tag?

ankushbkadam commented on src/Components/Home/index.tsx:
```diff
@@ -0,0 +1,20 @@
+   return (
+      <div>
+          <Carousel/>
+         <section className="container card-section pt-5 mb-4">
```
make a section component with these styles and classes, it should accept children component, lets discuss if you dont get it, it is for reusability

ankushbkadam commented on src/Components/Navbar/index.tsx:
```diff
@@ -2,27 +2,28 @@ import React from "react";
 import { Link, NavLink } from 'react-router-dom'
 import { ROUTES } from "../../Constants/routeConstants";
+import { IMGLINK } from "../../Constants/imagelinkConstant";
 const Navbar = () => {
```
keep a empty line after all imports, indent file properly

ankushbkadam commented on src/Components/Navbar/index.tsx:
```diff
@@ -2,27 +2,28 @@ import React from "react";
 import { ROUTES } from "../../Constants/routeConstants";
+import { IMGLINK } from "../../Constants/imagelinkConstant";
 const Navbar = () => {
+      
```
remove such unwanted spaces

ankushbkadam commented on src/Components/Navbar/index.tsx:
```diff
@@ -2,27 +2,28 @@ import React from "react";
         <nav className="navbar navbar-expand-md navbar-dark bg-dark sticky-top">
             <div className="container-fluid">
-                <a className="navbar-brand" href={ROUTES.HOME}>Navbar</a>
+                <img src={IMGLINK.logo1} className="navbar-brand" style={{width:"50px"}} />
```
inline styles? never use them

ankushbkadam commented on src/Components/Navbar/index.tsx:
```diff
@@ -2,27 +2,28 @@ import React from "react";
                 </button>
                 <div className="collapse navbar-collapse" id="navbarNavDropdown">
                     <ul className="navbar-nav">
                         <li className="nav-item">
```
li tag is repeating, why are we not using reusability feature

akshayyelvankar replied:
added reusable Listitem component

ankushbkadam commented on src/Constants/imagelinkConstant.ts:
```diff
@@ -0,0 +1,7 @@
+
+export const IMGLINK = {
```
if we are using all capital case, for readability we should give separator for example IMG_LINK

akshayyelvankar replied:
Changes Modified.

ankushbkadam commented on src/Constants/imagelinkConstant.ts:
```diff
@@ -0,0 +1,7 @@
+
+export const IMGLINK = {
+    pic1: 'https://cdn.pixabay.com/photo/2016/12/06/01/26/colour-1885352_960_720.jpg',
```
what is pic1 pic2, give meaningful name,
why constants are not in capital case?

akshayyelvankar replied:
name is modified pic to SLIDE.

ankushbkadam commented on src/Constants/routeConstants.ts:
```diff
@@ -1,4 +1,5 @@
-}
\ No newline at end of file
+     ABOUT:'/about',
+     CONVERTER:'/converter',
```
requested changes not added here, I asked you to use come meaningful name for converter, dont just use converter, as we will be having many converters later

ankushbkadam commented on src/App.css:
```diff
@@ -1,3 +1,82 @@
+  --gradient-text: linear-gradient(80.97deg, #AE67FA 1.84%, #F49867 102.67%);
+  --gradient-text: linear-gradient(80.97deg, #AE67FA -13.86%, #F49867 99.55%);
+}
+.gradient_bg {
```
keep empty line between two styles

akshayyelvankar replied:
empty line added.

ankushbkadam commented on src/App.css:
```diff
@@ -1,3 +1,82 @@
+  background: -o-radial-gradient(circle at 3% 25%, rgba(0, 41, 83, 1) 0%, rgba(4, 12, 24, 1) 100%);
+
+  /* ie 10+ */
+  background: -ms-radial-gradient(circle at 3% 25%, rgba(0, 41, 83, 1) 0%, rgba(4, 12, 24, 1) 100%);
```
dont just copy paste, are you using our app on ie anymore? keep only required browser code

akshayyelvankar replied:
removing unuseable browser support code.

ankushbkadam commented on src/App.css:
```diff
@@ -1,3 +1,82 @@
+}
+
+.converter-box {
+  display: flex;
```
we should give priority to bootstrap classes, for example display: flex we already have class for it, why are we not using it

ankushbkadam commented on src/App.css:
```diff
@@ -1,3 +1,82 @@
+  font-weight: 600;
+}
+
+.wrapper input {
```
it will apply to all inputs from all wrappers from our app, wrong way of styling, we should create new class and use it

ankushbkadam commented on src/Constants/imagelinkConstant.ts:
```diff
@@ -0,0 +1,7 @@
+
+export const IMGLINK = {
```
filename can be imageConstants 
or globalLinks (we can save all links from our app to one place)
note: use plural form not constant, it should be constants!

akshayyelvankar replied:
file name are modified.

ankushbkadam commented on src/Components/Navbar/index.tsx:
```diff
@@ -2,27 +2,28 @@ import React from "react";
 import { Route, Routes } from 'react-router';
 import { Link, NavLink } from 'react-router-dom'
```
remove all warnings we are getting, bad coding practices

akshayyelvankar replied:
warning removed.

ankushbkadam commented on src/Components/Navbar/index.tsx:
```diff
@@ -2,27 +2,28 @@ import React from "react";
         <nav className="navbar navbar-expand-md navbar-dark bg-dark sticky-top">
             <div className="container-fluid">
-                <a className="navbar-brand" href={ROUTES.HOME}>Navbar</a>
+                <img src={IMGLINK.logo1} className="navbar-brand" style={{width:"50px"}} />
```
why logo dont navigate to home page?

akshayyelvankar replied:
navigation added in logo.

akshayyelvankar committed:
some modified changes like Unwanted Spaces are Removed,added one line spaces after import,added fragments,renaming converter as a RGB_to_HEX,removing unuseable browser support code,warning remove,navigation added in logo.

akshayyelvankar committed:
changes modified

ankushbkadam commented on src/Constants/globalLinks.ts:
```diff
@@ -0,0 +1,6 @@
+export const IMG_LINK = {
+    SLIDE_1: 'https://cdn.pixabay.com/photo/2016/12/06/01/26/colour-1885352_960_720.jpg',
+    SLIDE_2: "https://cdn.pixabay.com/photo/2015/10/30/20/13/sunrise-1014712_960_720.jpg",
+    SLIDE_3: "https://cdn.pixabay.com/photo/2016/06/02/02/33/triangles-1430105_960_720.png",
```
SLIDE_1,2,3?
What if we have 100 photos, how to you identify what SLIDE_56 refers to?
This is why I asked you to give meaningful names.

Same for logo1

akshayyelvankar replied:
added some meaning full name like RGB_TEX,NIGHT_DARK, POLYGON_TEX,RGB_BRAND_LOGO

akshayyelvankar committed:
replacing SLIDE name to some meaningfull file name

akshayyelvankar committed:
some spaces removed

akshayyelvankar committed:
imported Link tag in Home component

akshayyelvankar committed:
removed all css of RGB-to-HEX component.

akshayyelvankar committed:
added reusable component in Navbar
